azurerm/internal/services/network/private_endpoint_resource.go

```diff
@@ -621,6 +628,12 @@ func flattenPrivateLinkEndpointServiceConnection(serviceConnections *[]network.P
 			if strings.HasSuffix(privateConnectionId, ".azure.privatelinkservice") {
 				attrs["private_connection_resource_alias"] = privateConnectionId
 			} else {
+				// There is a bug from ARM Cache, the PE created from portal could be with the connection id for postgresql server "Microsoft.DBForPostgreSQL" instead of "Microsoft.DBforPostgreSQL"
+				if strings.Contains(strings.ToLower(privateConnectionId), "microsoft.dbforpostgresql") {
+					if serverId, err := postgresqlParse.ServerID(privateConnectionId); err == nil {
+						privateConnectionId = serverId.ID()
+					}
+				}
```

Comment:
> all of the "DB For MySql/Postgres/MariaDB" etc resources have changed their ID's in the recent Azure SDK release (55.7.0 [example](https://github.com/Azure/azure-sdk-for-go/pull/15069/files#diff-8f45825717fb6d985585faad3684d04d3a1452a4176dcda445e729cd7c0e3f71L99-R98)) - so we may want to do this for all of those?

@tombuildsstuff, I totally agree with you, and I agree that that work needs to be done for all of the changed resource ID's, but I feel this fix should be ok to merge now to unblock current customers hitting this issue. Can we let this fix through now and then add the rest of the fixes for the other resources to our backlog to be fixed in a future release? TBH, there needs to be some investigation around this for each of the resources as there are two ways to pass this value to the private endpoint.

Reply:
From my knowledge of postgresql, they just change the request path. The id is always in "Microsoft.DBforXX" from a "GET" operation. The inconsistency comes from when the user create the resources by portal, it'll use its ARM resource id, the ARM resource ids are in wrong casing.
There will be no problems when using TF end to end. 
I'll ask more details to the service team.

Reply:
@WodansSon we'll merge this for this weeks release so that should be fine - but I think it'd be worth pre-emptively doing this for the other `DBFor` resources since we're doing that here - @yupwei68 would you be able to duplicate this for the MariaDB/MySQL too, since ID Parsers exist for those?

Reply:
I've updated for mysql and mariadb